vars/capsulePipelineWrapper.groovy

```diff
@@ -103,6 +115,83 @@ def call() {
+                        ].each { envName, flags ->
+                            sh label: "generate templates: ${envName}", script: """
+                                ./vegacapsule template genesis \
+                                    --home-path ${env.CONFIG_HOME} \
+                                    --path ${env.TEMPLATES_HOME}/genesis.tmpl.json \
+                                    ${flags}
+
+                                ./vegacapsule template node-sets \
+                                    --home-path ${env.CONFIG_HOME} \
+                                    --path ${env.TEMPLATES_HOME}/vega.validators.tmpl.toml \
+                                    --nodeset-group-name validators \
+                                    --type vega \
+                                    --with-merge \
+                                    ${flags}
+
+                                ./vegacapsule template node-sets \
+                                    --home-path ${env.CONFIG_HOME} \
+                                    --path ${env.TEMPLATES_HOME}/tendermint.validators.tmpl.toml \
+                                    --nodeset-group-name validators \
+                                    --type tendermint \
+                                    --with-merge \
+                                    ${flags}
+
+                                ./vegacapsule template node-sets \
+                                    --home-path ${env.CONFIG_HOME} \
+                                    --path ${env.TEMPLATES_HOME}/vega.full.tmpl.toml \
+                                    --nodeset-group-name full \
+                                    --type vega \
+                                    --with-merge \
+                                    ${flags}
+
+                                ./vegacapsule template node-sets \
+                                    --home-path ${env.CONFIG_HOME} \
+                                    --path ${env.TEMPLATES_HOME}/tendermint.full.tmpl.toml \
+                                    --nodeset-group-name full \
+                                    --type tendermint \
+                                    --with-merge \
+                                    ${flags}
+
+                                ./vegacapsule template node-sets \
+                                    --home-path ${env.CONFIG_HOME} \
+                                    --path ${env.TEMPLATES_HOME}/data_node.full.tmpl.toml \
+                                    --nodeset-group-name full \
+                                    --type data-node \
+                                    --with-merge \
+                                    ${flags}
+                            """
+                        }
+                    }
+                    sh label: "sync configs to s3", script: """
+                        aws s3 sync ${env.CONFIG_HOME}/ ${env.S3_CONFIG_HOME}/
+                    """
```

Comment:
I am worried about the case when the s3 upload fails. then the git commit won't happen.

I see a couple better or worse workarounds:
- make s3 upload and git commit separate Stages and run them in parallel,
- add retry for s3 sync (I think it should be safe to retry)
- wrap s3 upload in try-catch